File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/CircularProgressBar/CircularProgressBar.m
//
//  CircularProgressBar.m
//  sample-chat
//
//  Created by Injoit on 2/7/20.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "CircularProgressBar.h"

@interface CircularProgressBar()
@property (strong, nonatomic) CAShapeLayer *shapeLayer;
@end

@implementation CircularProgressBar

- (void)awakeFromNib {
    [super awakeFromNib];

    self.shapeLayer = [[CAShapeLayer alloc] init];
    UIBezierPath *circularPath = [UIBezierPath bezierPathWithArcCenter:CGPointZero radius:25.0f startAngle:0.0f endAngle:2*M_PI clockwise:YES];
    self.shapeLayer.path = circularPath.CGPath;
    UIColor *appColor = [UIColor colorWithRed:0.0f green:0.48f blue:1.0f alpha:1.0f];
    self.shapeLayer.strokeColor = appColor.CGColor;
    self.shapeLayer.lineWidth = 6.0f;
    self.shapeLayer.fillColor = UIColor.clearColor.CGColor;
    [self.shapeLayer setLineCap:kCALineCapRound];
    self.shapeLayer.position = [self convertPoint:self.center fromView:self.superview];
    self.shapeLayer.transform = CATransform3DMakeRotation(-M_PI/2, 0.0f, 0.0f, 1.0f);
    self.shapeLayer.strokeEnd = 0;
    [self.layer insertSublayer:self.shapeLayer atIndex:0];
}

//MARK: Public
- (void)setProgressTo:(CGFloat)progressConstant {
    if (progressConstant > self.shapeLayer.strokeEnd) {
        self.shapeLayer.strokeEnd = progressConstant;
    }
}

@end

File: sample-chat-swift/sample-chat-swift/Components/ChatScreen/Chat/CustomUI/CircularProgressBar/CircularProgressBar.swift
//
//  CircularProgressBar.swift
//  sample-chat-swift
//
//  Created by Injoit on 1/8/20.
//  Copyright © 2020 quickBlox. All rights reserved.
//

import UIKit

class CircularProgressBar: UIView {
    
    let shapeLayer = CAShapeLayer()
    private var pathCenter: CGPoint{get{ return self.convert(self.center, from:self.superview)}}
    
    //MARK: awakeFromNib
    override func awakeFromNib() {
    super.awakeFromNib()

        let circularPath = UIBezierPath(arcCenter: .zero, radius: 25, startAngle: 0, endAngle: 2 * CGFloat.pi, clockwise: true)
        shapeLayer.path = circularPath.cgPath
        shapeLayer.strokeColor = UIColor(named: "AppColor")?.cgColor
        shapeLayer.lineWidth = 6
        shapeLayer.fillColor = UIColor.clear.cgColor
        shapeLayer.lineCap = CAShapeLayerLineCap.round
        shapeLayer.position = pathCenter
        
        shapeLayer.transform = CATransform3DMakeRotation(-CGFloat.pi / 2, 0, 0, 1)
        
        shapeLayer.strokeEnd = 0
        layer.insertSublayer(shapeLayer, at:0)
    }

    //MARK: Public
    public func setProgress(to progressConstant: CGFloat) {
        shapeLayer.strokeEnd = progressConstant
    }
}
